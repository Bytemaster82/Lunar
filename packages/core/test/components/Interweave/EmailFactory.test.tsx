import React from 'react';
import { shallow } from 'enzyme';
import EmailFactory from '../../../src/components/Interweave/factories/Email';

describe('linkFactory()', () => {
  const emailParts = {
    host: '',
    username: '',
  };

  it('renders a link', () => {
    const wrapper = shallow(<EmailFactory emailParts={emailParts}>email@airbnb.com</EmailFactory>);

    expect(wrapper).toMatchSnapshot();
  });

  it('adds mailto: to emails', () => {
    const wrapper = shallow(<EmailFactory emailParts={emailParts}>email@airbnb.com</EmailFactory>);

    expect(wrapper.prop('href')).toBe('mailto:email@airbnb.com');
  });

  it('sets open in new window', () => {
    const wrapper = shallow(
      <EmailFactory emailParts={emailParts} newWindow>
        email@airbnb.com
      </EmailFactory>,
    );

    expect(wrapper.prop('openInNewWindow')).toBe(true);
  });

  it('sets small prop on link', () => {
    const wrapper = shallow(
      <EmailFactory emailParts={emailParts} newWindow small>
        email@airbnb.com
      </EmailFactory>,
    );

    expect(wrapper.prop('small')).toBe(true);
    expect(wrapper.prop('large')).toBe(false);
  });

  it('sets large prop on link', () => {
    const wrapper = shallow(
      <EmailFactory emailParts={emailParts} newWindow large>
        email@airbnb.com
      </EmailFactory>,
    );

    expect(wrapper.prop('small')).toBe(false);
    expect(wrapper.prop('large')).toBe(true);
  });
});
