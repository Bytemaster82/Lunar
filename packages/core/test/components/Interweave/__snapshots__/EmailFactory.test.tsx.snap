// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`linkFactory() renders a link 1`] = `
<Link
  baseline={true}
  href="mailto:email@airbnb.com"
  large={false}
  openInNewWindow={false}
  small={false}
>
  email@airbnb.com
</Link>
`;
